### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by yohanboniface**
Very basic version, to have a start :)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Usine is a deployement tool that tries to make as little as possible for you not
+to stay on your way.
```
Comment (davidbgk) [inline docs/index.md]: I think the formulation is confusing here, maybe @vinyll can make a suggestion, I'm a bit out of idea.

File: docs/reference.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+## run(cmd)
+
+This is the main helper, which basically runs any command on the remote server.
+
+##### Arguments
+
+- **cmd**: the actual command to be run
+
+Returns as `Status` instance.
```
Comment (davidbgk) [inline docs/reference.md]: a? (vs. as)

File: docs/reference.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+## mkdir(path, parents=True, mode=None)
+
+Run `mkdir` command on the remote server to create a directory. See `man mkdir`
+for more info about the command.
+
+##### Arguments:
+
+- **path**: the path of the directory
+- **parents** (default: True): also create parents directory if needed and do
```
Comment (davidbgk) [inline docs/reference.md]: createS?

File: docs/reference.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+## chown(mode, path, recursive=True, preserve_root=True)
+
+Run `chown` command in the remote server to change file or directory ownership.
+See `man chown` for more info.
+
+##### Arguments
+
+- **mode**: `user:group` to changer the ownership to
+- **path**: the path of the file or directory
+- **recursive** (default: True): operate on files and directories recursively
```
Comment (davidbgk) [inline docs/reference.md]: operateS?

File: usine.py [hunk 5]
```diff
@@ -305,6 +313,11 @@ def execute(self, cmd, **kwargs):
             sys.exit(ret.code)
         return ret
 
+    def __call__(self, cmd, **kwargs):
```
Comment (davidbgk) [inline usine.py:335]: Clever 😎

**Review by davidbgk — APPROVED** [review]

File: docs/reference.md (hunk 4 above)
Comment (yohanboniface) [reply docs/reference.md]: Good question :p
It's never clear for me.
This sentence is actually a copy-paste from `man chown`, and there I see they never use the "s".
So maybe I should instead align on that way everywhere?

Comment (davidbgk) [reply docs/reference.md]: No strong opinion about that, do what is consistent over code and I'm OK :)

Comment (yohanboniface) [reply docs/reference.md]: I've hesitated but eventually removed all the "S".

File: docs/how-to.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
+# How to guides
```
Comment (vinyll) [inline docs/how-to.md]: _How-to guide_?

File: docs/how-to.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+if __name__ == '__main__':
+    with connect(hostname='myhost'):
+        minici.run()
+```
+
+And now you can run it like this:
+
+    myscript.py foo --force
+
+If you want to expose the hostname in the command line (or any other global
```
Comment (vinyll) [inline docs/how-to.md:57]: `confipath`? Like `confiture` but with _path_?

File: docs/how-to.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+variable like the `confipath`), here is a bit more complete example:
```
Comment (vinyll) [inline docs/how-to.md]: _a more complete_

File: docs/index.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
+Usine is a minimalist and pythonnic deployment tool.
```
Comment (vinyll) [inline docs/index.md]: _python**n**ic_?

File: docs/index.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+It that tries to do as little as possible for you not
```
Comment (vinyll) [inline docs/index.md]: _It will do as little as possible not to be in the way_?

File: docs/index.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+- manages SSH connection (with [paramiko](http://paramiko.org/))
+- manages configuration files
+- allows you to run commands in a remote server
+- wraps some common commands for a more pythonnic API (and thus ease the use of
+  `*args` and `**kwargs` patterns)
+- exposes some pythonic helpers to manage env vars, sudo and such
+
+
+## Install
+
```
Comment (vinyll) [inline docs/index.md]: extra newline?

File: docs/index.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+Here is what a very basic script would look like:
+
+```python
+from usine import run, connect
+
+with connect(hostname='me@remote'):
+    run('ls /tmp/foobar')
+```
+
+## When to use it
```
Comment (vinyll) [inline docs/index.md]: question mark?

File: docs/index.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+
+When you have a few servers to maintain and you want to be sure every
+needed command is documented and reproductible by you and your comates.
+
+
+## When not to use it
```
Comment (vinyll) [inline docs/index.md]: question mark?

File: docs/reference.md [hunk 14]
```diff
@@ -0,0 +1,217 @@
...
+
+## mkdir(path, parents=True, mode=None)
+
+Run `mkdir` command on the remote server to create a directory. See `man mkdir`
+for more info about the command.
+
+##### Arguments:
+
+- **path**: the path of the directory
+- **parents** (default: True): also create parents directory if needed and do
```
Comment (vinyll) [inline docs/reference.md]:
```
`True`
```

File: docs/reference.md [hunk 15]
```diff
@@ -0,0 +1,217 @@
...
+- **kwargs**: key/value pair to set environment variables
+
+
+## screen(name)
+
+Run the command inside a screen.
+
+##### Arguments
+
+- **name** (default: `usine`) the name of the screen to be created
```
Comment (vinyll) [inline docs/reference.md:229]: Here you are formatting the default value as code, which you weren't previously (like with `True`)

**Pull request merged**